src/test/java/org/gridsuite/study/server/StudyTest.java

```diff
@@ -2577,24 +2592,67 @@ public void testLineSplitWithVoltageLevel() {
         assertEquals(lineSplitWoVLasJSON, updateRequest.get().getBody());
 
         webTestClient.post()
-            .uri("/v1/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/line-splits",
-                studyNameUserIdUuid, modificationNodeUuid)
-            .bodyValue("bogus")
-            .exchange()
-            .expectStatus().is5xxServerError()
-            .expectBody(String.class).isEqualTo("400 BAD_REQUEST");
+                .uri("/v1/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/line-splits",
+                        studyNameUserIdUuid, modificationNodeUuid)
+                .bodyValue("bogus")
+                .exchange()
+                .expectStatus().is5xxServerError()
+                .expectBody(String.class).isEqualTo("400 BAD_REQUEST");
 
         webTestClient.put()
-            .uri("/v1/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/modifications/{modificationUuid}/line-splits",
-                studyNameUserIdUuid, modificationNodeUuid, MODIFICATION_UUID)
-            .bodyValue("bogus")
-            .exchange()
-            .expectStatus().is5xxServerError()
-            .expectBody(String.class).isEqualTo("400 BAD_REQUEST");
+                .uri("/v1/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/modifications/{modificationUuid}/line-splits",
+                        studyNameUserIdUuid, modificationNodeUuid, MODIFICATION_UUID)
+                .bodyValue("bogus")
+                .exchange()
+                .expectStatus().is5xxServerError()
+                .expectBody(String.class).isEqualTo("400 BAD_REQUEST");
 
         requests = getRequestsWithBodyDone(2);
     }
 
+    @SneakyThrows
+    @Test
+    public void testLineAttachToVoltageLevel() {
+        createStudy("userId", CASE_UUID);
+        UUID studyNameUserIdUuid = studyRepository.findAll().get(0).getId();
+        UUID rootNodeUuid = getRootNodeUuid(studyNameUserIdUuid);
+        NetworkModificationNode modificationNode = createNetworkModificationNode(studyNameUserIdUuid, rootNodeUuid);
+        UUID modificationNodeUuid = modificationNode.getId();
+
+        String createVoltageLevelAttributes = "{\"voltageLevelId\":\"vl1\",\"voltageLevelName\":\"voltageLevelName1\""
+                + ",\"nominalVoltage\":\"379.1\", \"substationId\":\"s1\"}";
+
+        String createLineAttributes = "{\"seriesResistance\":\"25\",\"seriesReactance\":\"12\"}";
+
+        String createLineAttachToVoltageLevelAttributes = "{\"lineToAttachToId\": \"line3\", \"percent\":\"10\", \"mayNewVoltageLevelInfos\":" +
+                createVoltageLevelAttributes + "\"attachmentLine\":\"" + createLineAttributes + "\"}";
```

Comment:
Maybe better : creating the attributes objects, and then serialize them to json ?

Reply:
This would necessitate to create specific builders in test package. Since the number of attributes to create is not too big, does not seem necessary to use builders.